### PR TITLE
Fix file picker not respecting hidden files settings

### PR DESCRIPTION
**Pull request by CarlSchwan**
This will only respect the setting inside the file app. For other apps
we will either need to do an API call or add an input field with the
same idea to spare a blocking api call.

Related to #30589

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1: good enough as a quick fix for files app

hopefully we can move this to initial states later on

File: dist/settings-apps-settings-apps.js [hunk 1]
```diff
@@ -3610,4 +3610,8 @@ render._withStripped = true
 /***/ })
 
 }]);
-//# sourceMappingURL=settings-apps-settings-apps.js.map?v=a5f2375a4af256153b26
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (tcitworld) [inline dist/settings-apps-settings-apps.js:3613]: Git conflicts

Comment (CarlSchwan) [reply dist/settings-apps-settings-apps.js]: Just did a 

```
rm -rf dist
npm ci && npm run build
```

and now some files got removed. I guess the webpack targets were previously deleted but someone forgot to also delete the files from the dist folder.

Comment (artonge) [reply dist/settings-apps-settings-apps.js]: Will be fixed by: https://github.com/nextcloud/server/pull/30605

Comment (artonge) [reply dist/settings-apps-settings-apps.js]: Will be fixed by: https://github.com/nextcloud/server/pull/30605

**Review by artonge — APPROVED** [review]

**Comment by CarlSchwan** [comment]
@PVince81 Can you re-review now that there is some new tests? :D

**Comment by CarlSchwan** [comment]
actually the PR with the new tests is https://github.com/nextcloud/server/pull/30600/files. Sorry messed up my tabs

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by CarlSchwan** [comment]
Manual backports:

* https://github.com/nextcloud/server/pull/30677
* https://github.com/nextcloud/server/pull/30673
* https://github.com/nextcloud/server/pull/30672

**Pull request merged**